lib/ninja.py

```diff
@@ -57,9 +57,12 @@ class _OutputAccumulator:
 
     @staticmethod
     def get_tty_width():
+        tput_cmd = ["tput", "cols"]
+        if os.getenv("TERM") is None:
+            tput_cmd.extend(["-T", "linux"])
```

Comment:
I'm not sure if this is a good idea. This is going to look up the terminal width in the 'linux' terminfo settings, but I don't see why this would give us an accurate number if `TERM` and `COLUMNS` aren't defined. For example in CI, it's not at all clear to me that whatever the '`linux`' entry is will correspond to the actual width of the CI logging system.

My suggestion is: run the command, and if it fails, return a safe and sensible default like 70.

Reply:
For Ubuntu environments in both AWS CodeBuild and GitHub Actions the`tput cols -T linux` command will return 80.

Reply:
I created a workflow in GitHub Actions to test this out ([link](https://github.com/angelonakos/aws-build-accumulator/runs/4928326871?check_suite_focus=true)).

Reply:
Right, but as noted before, this is not robust for systems that don't know what a `linux` terminal is, including macOS apparently...

Reply:
I also tested in an AWS CodeBuild project that ran in a Linux environment type. 

Specifically, if one takes the below code, places it in a directory called `foo`, compresses it  into a .zip, uploads it to an S3 bucket, creates an AWS CodeBuild project running on Linux, whose buildspec takes the single command of `python3 foo/foo.py` , and finally attempts to start a build, they will notice that the build succeeds and AWS CodeBuild printing out in the build logs `there was an output of 80`. This the code that I mentioned:


```python
#!/usr/bin/env python3

import logging
import os
import subprocess


def main():
    tput_cmd = ["tput", "cols"]
    if os.getenv('TERM') is None:
        tput_cmd.extend(["-T", "linux"])
    try:
        proc = subprocess.run(
            tput_cmd, text=True, stdout=subprocess.PIPE, check=True)
    except (FileNotFoundError, subprocess.CalledProcessError):
        return None
    if proc.returncode:
        logging.info("there was a return code of %s", proc.returncode)
        return None
    try:
        _output = proc.stdout.strip()
        logging.info("there was an output of %s", _output)
        return int(_output)
    except TypeError:
        return None


if __name__ == "__main__":
    logging.basicConfig(format="%(message)s", level=logging.INFO)
    main()

```